delivery_tnt_oca/models/delivery_carrier.py

```diff
@@ -0,0 +1,170 @@
+        compute="_compute_tnt_product_service", string="Product service",
+    )
+    tnt_product_service_d = fields.Selection(
+        selection=[
+            ("EX", "Express"),
+            ("EX09", "9:00 Express"),
+            ("EX10", "10:00 Express"),
+            ("EX12", "12:00 Express"),
+        ],
+        default="EX",
+        string="Product service (Docs)",
+    )
+    tnt_product_service_n = fields.Selection(
+        selection=[
+            ("EC", "Economy Express"),
+            ("EC12", "12:00 Economy Express"),
+            ("EX", "Express"),
+            ("EX09", "9:00 Express"),
+            ("EX10", "10:00 Express"),
+            ("EX12", "12:00 Express"),
+        ],
+        default="EX",
+        string="Product service (Non docs)",
+    )
+    tnt_payment_indicator = fields.Selection(
+        selection=[("S", "Sender pays"), ("R", "Receiver pays")],
+        default="S",
+        string="Payment indicator",
+    )
+    tnt_line_of_business = fields.Selection(
+        selection=[
+            ("1", "Domestic dk to dk transfers"),
+            ("2", "international non-domestic transfers"),
+        ],
+        default="1",
+        string="Line of business",
+    )
+    tnt_collect_time_from = fields.Char(default="10:30", string="Collect time from",)
+    tnt_collect_time_to = fields.Char(default="16:00", string="Collect time to",)
+
+    def _compute_tnt_product_code(self):
+        for item in self:
+            if item.tnt_product_type == "D":
+                item.tnt_product_code = item.tnt_product_code_d
+            else:
+                item.tnt_product_code = item.tnt_product_code_n
+
+    def _compute_tnt_product_service(self):
+        for item in self:
+            if item.tnt_product_type == "D":
+                item.tnt_product_service = item.tnt_product_service_d
+            else:
+                item.tnt_product_service = item.tnt_product_service_n
+
+    def tnt_oca_rate_shipment(self, order):
+        tnt_request = TntRequest(self, order)
+        response = tnt_request.rate_shipment()
+        if response["success"]:
+            response["price"] = self._tnt_oca_get_response_price(
+                response, order.currency_id, order.company_id
+            )
+        return {
+            "success": response["success"],
+            "price": response["price"],
+            "error_message": False,
+            "warning_message": False,
+        }
+
+    def _tnt_oca_get_response_price(self, response, currency, company):
+        """We need to convert the price if the currency is different."""
+        price = float(response["price"])
+        if response["currency"] != currency.name:
+            price = currency._convert(
+                price,
+                self.env["res.currency"].search([("name", "=", response["currency"])]),
+                company,
+                fields.Date.today(),
+            )
+        return price
+
+    def tnt_oca_action_label(self, picking):
```

Comment:
Again should be private?

```suggestion
    def _tnt_oca_action_label(self, picking):
```